### DISCUSSION TITLE
i am not able to push the code from local repo to remote repo

### DISCUSSION THREAD

**Discussion by sanagalasrikanthreddy** [community/community#30612] [New to GitHub]
hello 
i am getting this kind of message . i use git push command then i enter username and password of git hub but i getting the below.
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/sanagala-ec-mnc/amazon.git/'
what can i do please healp me 
thanks 
srikanth

**Comment by airtower-luna** [comment]
Like the message says you cannot use your password for Git operations. You need to [create a personal access token (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and use that instead of your password. Your token needs to have the **repo** scope, and also **workflow** if you use Actions.

**Reply by jaxkashif34** [reply]
i faced the same error few days ago this stackover flow answer help me...

[https://stackoverflow.com/questions/68775869/message-support-for-password-authentication-was-removed-please-use-a-personal]

**Comment by JohnSmithW** [comment]
In Github, go to Settings -> Developer settings(At the bottom of left sidebar) -> Personal access tokens (Left sidebar) -> Generate new token, after you generate it, save it somewhere on your PC and use it as a password whenever you need to authorize via Github.
